devopstools-jenkins-chefhabitat-terraform/azuredeploy.json

```diff
@@ -0,0 +1,391 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "username for the virtual machines."
+            }
+        },
+        "authenticationType": {
+            "type": "string",
+            "defaultValue": "",
+            "allowedValues": [
+                "sshPublicKey"
+            ],
+            "metadata": {
+                "description": "authentication type for the virtual machines."
+            }
+        },
+        "password":{
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "password for elk,vmss and mongodbterraform virtual machines"
+            }
+        },
+        "sshPublicKey": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "generate ssh public key"
+            }
+        },
+        "jenkinsVmSize": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "vm size"
+            }
+        },
+        "buildInstanceVmSize": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "vm size"
+            }
+        },
+        "azureApplicationId": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "ad application id"
+            }
+        },
+        "azureClientSecret": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "ad clientsecret"
+            }
+        },
+        "_artifactsLocation": {
```

Comment:
see the BP guide for how to handle artifacts - you need a defaultValue and a sasToken